scripts/download_boxes.rb

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env ruby
+
+require 'json'
+require 'getoptlong'
+require 'net/http'
+require 'progressbar'
+require 'fileutils'
+
+PROVIDER = 'provider'
+PLATFORM = 'platform'
+PLATFORM_VERSION = 'platform_version'
+BOX = 'box'
+
+opts = GetoptLong.new(
+    [ '--dir', '-d', GetoptLong::REQUIRED_ARGUMENT ],
+    [ '--boxes-dir', '-b', GetoptLong::REQUIRED_ARGUMENT ],
+    [ '--force', '-f', GetoptLong::NO_ARGUMENT ],
+    [ '--help', '-h', GetoptLong::NO_ARGUMENT ]
+)
+
+dir = nil
+boxes_dir = nil
+
+force = false
+
+opts.each do |opt, arg|
```

Comment:
All code which is not a constant definition must be placed in functions.
- main
- parse files
- download image
- path list storing